tlatools/org.lamport.tlatools/src/tlc2/debug/TLCSourceBreakpoint.java

```diff
@@ -55,8 +58,9 @@ public TLCSourceBreakpoint(final String module, final SourceBreakpoint s) {
 			}
 		}
 		hits = h;
+		this.condition = TLCBreakpointExpression.process(semanticRoot, s.getCondition());
```

Comment:
This needs to check if s.getCondition is an existing definition such as `TypeOK`, ... that doesn't need processing.

Reply:
If the user puts in an existing definition like `TypeOK` I think the processing code will ultimately result in what the user intends; it could be short-circuited by looking up existing definitions in the module but this sort of special casing would also add another code path to test.

Reply:
How does it ultimately result in what the user intends?  Does it create a module containing a single definition, `TypeOK_139488708`, which delegates to `TypeOK` in the original specification, similar to how it would handle true ad-hoc/on-the-fly/dynamic breakpoint expressions?

Reply:
It would create something like
```tla
---- MODULE __BreakpointModule__12345678
EXTENDS OriginalModuleName
__BreakpointDef__12345678(i, j, k) == TypeOK
====
```
and then when the breakpoint is evaluated it should just evaluated `TypeOK`.